### ISSUE TITLE
[iOS] Crash after scrolling and zooming

### ISSUE THREAD

**Issue by picciano** [mapbox/mapbox-gl-native#1029]
Using iPhone 5s running iOS 8.2
Default map style, no other changes.
Version information: https://github.com/mapbox/mapbox-gl-native/commit/b4dbba93f8

This seems to happen consistently after scrolling and zooming in and out for about 30 seconds to a minute. This is most often seen zooming out after having zoomed in.

```
Thread 9 name:  Map
Thread 9 Crashed:
0   AGXGLDriver                   	0x0000000182552550 agxuEndRenderCommand(AGXContextRec*, AGXHalPrimitiveState*) + 348
1   AGXGLDriver                   	0x0000000182552538 agxuEndRenderCommand(AGXContextRec*, AGXHalPrimitiveState*) + 324
2   AGXGLDriver                   	0x00000001825519d4 SubmitPackets(AGXContextRec*) + 44
3   GLEngine                      	0x0000000187904dc8 gleFreeOrphan + 120
4   GLEngine                      	0x0000000187904a1c gleCleanupOrphans + 40
5   GLEngine                      	0x0000000187904934 gleOrphanBufferObject + 396
6   GLEngine                      	0x00000001878aad20 glDeleteBuffers_Exec + 1740
7   TallyGo                       	0x00000001000fc3b8 mbgl::VectorTileData::~VectorTileData() (buffer.hpp:24)
8   libc++.1.dylib                	0x000000019454d760 std::__1::__shared_weak_count::__release_shared() + 60
9   TallyGo                       	0x00000001000eb268 mbgl::Source::update(mbgl::Map&, uv::worker&, mbgl::util::ptr<mbgl::Style>, mbgl::GlyphAtlas&, mbgl::GlyphStore&, mbgl::SpriteAtlas&, mbgl::util::ptr<mbgl::Sprite>, mbgl::TexturePool&, std::__1::function<void ()>) (source.cpp:346)
10  TallyGo                       	0x00000001000e7938 mbgl::Map::updateTiles() (map.cpp:697)
11  TallyGo                       	0x00000001000e6cb0 mbgl::Map::prepare() (map.cpp:743)
12  TallyGo                       	0x0000000100168d1c uv__async_event (async.c:80)
13  TallyGo                       	0x0000000100169140 uv__async_io (async.c:156)
14  TallyGo                       	0x0000000100174834 uv__io_poll (kqueue.c:233)
15  TallyGo                       	0x00000001001697b8 uv_run (core.c:317)
16  TallyGo                       	0x00000001000e687c mbgl::Map::run() (map.cpp:256)
17  TallyGo                       	0x00000001000e7de8 void* std::__1::__thread_proxy<std::__1::tuple<mbgl::Map::start(bool)::$_4> >(void*) (thread:175)
18  libsystem_pthread.dylib       	0x00000001955bbdc4 _pthread_body + 160
19  libsystem_pthread.dylib       	0x00000001955bbd20 _pthread_start + 156
20  libsystem_pthread.dylib       	0x00000001955b8ef4 thread_start + 0
```

**Comment by picciano** [comment]
Could be the same as https://github.com/mapbox/mapbox-gl-native/pull/1016 ?

**Comment by picciano** [comment]
Fixed with #1016
